src/main/java/hudson/plugins/analysis/core/BuildHistory.java

```diff
@@ -187,11 +190,27 @@ private boolean isSuccessfulAction(final ResultAction<? extends BuildResult> act
         return null;
     }
 
-    private boolean hasValidResult(final AbstractBuild<?, ?> build) {
+    /**
+     * Added for backward compatibility. It generates <pre>AbstractBuild getReferenceBuild()</pre> bytecode during the build
+     * process, so old implementations can use that signature.
+     * 
+     * @see {@link WithBridgeMethods}
+     */
+    @Deprecated
+    private Object getReferenceAbstractBuild(Run run, Class targetClass) {
```

Comment:
This method is only called internally (linked to `@WithBrigdeMethods`),  I'm not sure if making the parameters final is required or not. Doing it anyway (in order to follow the style guide mainly).